File: complaint-management/mobile/Documentation/Complain_Management_System_Classes/lib/models/complaint/complaint.dart
import 'complaint_category.dart';
import 'complaint_priority.dart';
import 'complaint_status.dart';

class Complaint {
  String complaintId;
  String title;
  String description;
  ComplaintCategory category;
  ComplaintPriority priority;
  ComplaintStatus status;
  DateTime submittedDate;
  DateTime lastUpdated;
  DateTime? resolvedDate;
  String submittedBy;
  String? assignedTo;
  List<String> attachments;

  Complaint({
    required this.complaintId,
    required this.title,
    required this.description,
    required this.category,
    required this.priority,
    required this.status,
    required this.submittedDate,
    required this.lastUpdated,
    this.resolvedDate,
    required this.submittedBy,
    this.assignedTo,
    required this.attachments,
  });

  bool updateStatus(ComplaintStatus newStatus) {
   
    return false;
  }

  bool assignTo(String userId) {
 
    return false;
  }

  bool addAttachment(String filePath) {
   
    return false;
  }
}

File: complaint-management/mobile/Documentation/Classes_kotlin/models/complaint/Complaint.kt
package com.cms.models.complaint

import com.cms.enums.ComplaintCategory
import com.cms.enums.ComplaintPriority
import com.cms.enums.ComplaintStatus

data class Complaint(
    val id: String,
    val title: String,
    val description: String,
    val category: ComplaintCategory,
    val priority: ComplaintPriority,
    val status: ComplaintStatus,
    val createdByUserId: String,
    val createdAt: String
)
